examples/button.rs

```diff
@@ -0,0 +1,591 @@
+                        ButtonWidgetBundle::new(Style {
+                            size: Size::new(Val::Px(40.0), Val::Px(40.0)),
+                            margin: UiRect::all(Val::Px(10.0)),
+                            justify_content: JustifyContent::Center,
+                            align_items: AlignItems::Center,
+                            ..default()
+                        }, 
+                        BUTTON_THEME
+                    )).with_children(| button | {
+                        button.spawn_bundle(TextBundle::from_section(
+                            Icon::Wifi.to_string(),
+                            TextStyle {
+                                font: asset_server.load(MATERIAL_FONT).into(),
+                                font_size: BUTTON_FONT_SIZE,
+                                color: COLOR_TEXT,
+                            })
+                        );
+                    });
+
+                    // 2. Icon button
+                    example_showcase.spawn_bundle(
+                        ButtonWidgetBundle::new(Style {
+                            size: Size::new(Val::Px(40.0), Val::Px(40.0)),
+                            margin: UiRect::all(Val::Px(10.0)),
+                            justify_content: JustifyContent::Center,
+                            align_items: AlignItems::Center,
+                            ..default()
+                        }, 
+                        BUTTON_THEME
+                    )).with_children(| button | {
+                        button.spawn_bundle(TextBundle::from_section(
+                            Icon::Subtitles.to_string(),
+                            TextStyle {
+                                font: asset_server.load(MATERIAL_FONT).into(),
+                                font_size: BUTTON_FONT_SIZE,
+                                color: COLOR_TEXT,
+                            })
+                        );
+                    });
+
+                    // 3. Icon button
+                    example_showcase.spawn_bundle(
+                        ButtonWidgetBundle::new(Style {
+                            size: Size::new(Val::Px(40.0), Val::Px(40.0)),
+                            margin: UiRect::all(Val::Px(10.0)),
+                            justify_content: JustifyContent::Center,
+                            align_items: AlignItems::Center,
+                            ..default()
+                        }, 
+                        BUTTON_THEME
+                    )).with_children(| button | {
+                        button.spawn_bundle(TextBundle::from_section(
+                            Icon::Delete.to_string(),
+                            TextStyle {
+                                font: asset_server.load(MATERIAL_FONT).into(),
+                                font_size: BUTTON_FONT_SIZE,
+                                color: COLOR_TEXT,
+                            })
+                        );
+                    });
```

Comment:
In the demo you showed it looked like the centering didn't work for the "Delete" button (and possibly the other icon buttons as well). Not sure why though

Reply:
The buttons with both icon _and_ text were specifically _not_ centred, but on a closer look I think they should be.

Reply:
I think it's best if the icon is always at the start, but the text is centered in the rest of the button. Not sure how easy that would be though :)

Reply:
I think that isn't going to be super simple. Also there are times where you want the icon to be on either start or end for clarity reasons: See reference: https://mui.com/material-ui/react-button/#buttons-with-icons-and-label

Reply:
Also, RTL writing directions 🙃

Reply:
RTL should be supported for free if you use Flexbox to stack the icon and text :)

Reply:
Oh, right! In that case it should be simple :)